[Gilbb] iP

Gilbb opened the pull request:
## Cookie
### Cookie is your **personal** chatbot for managing your tasks
>“Keep calm and eat cookies.” - Cookie Monster ([source](https://www.goodreads.com/author/quotes/17927147.Cookie_Monster))

+ ~~Easy~~ VERY EASY to use with a GUI
+ Easy to remember commands
+ :bangbang: **IT'S FREE** :bangbang:

All you need to do is,
1. Download [cookie.jar](https://github.com/Gilbb/ip/releases/tag/Level-10).
2. Copy or move the jar file into an empty folder.
3. Open the terminal in that folder.
4. Run the command java -jar "cookie.jar"

Features:

- [ ] Managing todos, deadlines and events
- [ ] Save your tasks between sessions
---
Here's the entry point of the program, the <span style="color: red;">`Launcher`</span> class: 
```java
public class Launcher {
    public static void main(String[] args) {
        Application.launch(Main.class, args);
    }
}
```

Commit:
Add Gradle support

Eclipse-Dominator committed:
Bump gradle and lib version

aureliony committed:
build.gradle: Prevent generating a second JAR file

In build.gradle, the dependencies on distZip and/or distTar causes
the shadowJar task to generate a second JAR file for which the
mainClass.set("seedu.duke.Duke") does not take effect.
Hence, this additional JAR file cannot be run.
For this product, there is no need to generate a second JAR file
to begin with.

Let's remove this dependency from the build.gradle to prevent the
shadowJar task from generating the extra JAR file.

Gilbb committed:
Add greet and exit texts

Gilbb committed:
Add echo functionality

Gilbb committed:
Add list functionality

Gilbb committed:
Add mark and unmark functionality

Gilbb committed:
Add support for tracking ToDos, Deadlines, Events

Gilbb committed:
Add automated text ui testing

Gilbb committed:
Add error handling for some incorrect inputs

Gilbb committed:
Add support for deleting tasks from list

Gilbb committed:
Add support to save tasks into .txt and load file on start up

Gilbb committed:
Edit import statements in Cookie class

Gilbb committed:
Add support for LocalDate input for deadlines and events

Gilbb committed:
Refactor code and create new classes

Gilbb committed:
Move classes into Cookie package

Gilbb committed:
Merge branch 'add-gradle-support'

Gilbb committed:
Implement JUnit test for task classes

Gilbb committed:
Package app as executable JAR

Gilbb committed:
Add Javadoc comments to public methods

Gilbb committed:
Tweak code to comply with coding standard

Gilbb committed:
Merge branch 'branch-A-CodingStandard'

Gilbb committed:
Add find command for searching task by keyword

Gilbb committed:
Merge branch 'branch-Level-9'

Gilbb committed:
Fix spacing in Ui class

Gilbb committed:
Add checkstyle to project

Gilbb committed:
Add checkstyle to project and remove .idea file

Gilbb committed:
Add a GUI to Cookie

yihuididi commented on src/main/java/Cookie/Cookie.java:
```diff
@@ -0,0 +1,188 @@
+package Cookie;
```
I think package should be in all lower caps.

yihuididi commented on src/main/java/Cookie/Cookie.java:
```diff
@@ -0,0 +1,188 @@
+            String command = parser.parseCommand(input);
+            String description = parser.parseDescription(input);
+
+            switch (command) {
+            case "bye":
+                return handleQuit();
+            case "list":
+                return handlePrintTasks();
+            case "find":
+                return handleFind(description);
+            case "delete":
+                return handleDelete(description);
+            case "mark":
+                return handleMark(description);
+            case "unmark":
+                return handleUnmark(description);
+            case "todo":
+                return handleTodo(description);
+            case "deadline":
+                return handleDeadline(description);
+            case "event":
+                return handleEvent(description);
+            default:
+                throw new CookieException("Cookie does not understand this command. I'm sorry!!");
+
+            }
```
Good job on following the Coding Standards for Switch statements!

yihuididi commented on src/main/java/Cookie/Cookie.java:
```diff
@@ -0,0 +1,188 @@
+            throw new CookieException("The task you want to delete does not exist");
+        }
+        String response = ui.printDeleteTask(this.taskList.getTask(deleteIndex));
+        this.taskList.delete(deleteIndex);
```
I think no need to specify 'this' because there is no sharing of variable name.

yihuididi commented on src/main/java/Cookie/Cookie.java:
```diff
@@ -0,0 +1,188 @@
+
+        if (string.isEmpty()) {
+            throw new CookieException("Cookie does not know which task to delete.\n "
+                    + "(Please enter an integer after \"delete\")");
```
Good job for adding 8 whitespaces for long lines!

yihuididi reviewed (commented):
Overall Good Job! Just some nits to pick like having all small caps package names and using the fully qualified name for variables (e.g. using this.description).

W3iZhi commented on src/main/java/Cookie/Cookie.java:
```diff
@@ -0,0 +1,188 @@
+
+    private String handleUnmark(String string) throws CookieException {
+        if (string.isEmpty()) {
+            throw new CookieException("Cookie does not know which task to mark.\n "
```
Great job separating long lines

W3iZhi commented on src/main/java/Cookie/Cookie.java:
```diff
@@ -0,0 +1,188 @@
+package Cookie;
+
+import java.io.FileNotFoundException;
```
Good ordering of the imports

W3iZhi commented on src/main/java/Cookie/Cookie.java:
```diff
@@ -0,0 +1,188 @@
+    private String handleEvent(String string) throws CookieException {
+        String[] eventDetails = parser.parseEvent(string);
+        if (eventDetails.length < 2 || eventDetails[0].isEmpty()
+                || eventDetails[1].isEmpty() || eventDetails[2].isEmpty()) {
```
Great job of indentation

W3iZhi commented on src/main/java/Cookie/Cookie.java:
```diff
@@ -0,0 +1,188 @@
+            String description = parser.parseDescription(input);
+
+            switch (command) {
+            case "bye":
```
Good usage of switch case indents

W3iZhi reviewed (commented):
LGTM

Gilbb committed:
Add java assertions to Cookie class

Currently the program runs without checking for programming mistakes.

With assertions, programmer mistakes can be easily identified and bugs
can be fixed early.

Gilbb committed:
Merge pull request #1 from Gilbb/A-Assertions

Add java assertions to Cookie class

Gilbb committed:
Add java assertions to Cookie class
Currently the program runs without checking for programming mistakes.

With assertions, programmer mistakes can be easily identified and bugs
can be fixed early.

Gilbb committed:
Merge pull request #2 from Gilbb/branch-A-Assertions

Add java assertions to Cookie class

Gilbb committed:
Refactor command handling and input parsing

The `Cookie` class currently handles multiple command types directly, which makes it complex
and difficult to maintain. The `Parser` class handles minimal parsing of user input, leading
less modular codebase.

This approach complicates command management and makes the code harder to extend or modify.
Centralizing command logic and improving modularity will simplify maintenance and improve code readability.

- Move command handling logic from the `Cookie` class to new command classes (`ByeCommand`, `DeadlineCommand`,
    `DeleteCommand`, `EventCommand`, `FindCommand`, `ListCommand`, `MarkCommand`, `ToDoCommand`, `UnmarkCommand`).
- Refactor the `Parser` class to handle all input parsing and convert user inputs into command objects.
- Update the `Cookie` class to use the `Parser` class for command instantiation and delegate command execution to the
    newly created command classes.

Isolate command-specific logic into dedicated classes to improve modularity and separation of concerns. Centralize input
parsing in the `Parser` class to streamline command creation and ensure a single point of input processing.

This refactoring also includes more comprehensive JavaDoc header comments.

Gilbb committed:
Merge pull request #3 from Gilbb/branch-A-CodeQuality

Refactor command handling and input parsing

Gilbb committed:
Add assertions to Deadline, Event, Find, and ToDo

Currently, the Deadline, Event, Find, and ToDo commands accept user inputs without enforcing
any assertions to validate key fields like task descriptions or deadlines. This leaves potential room
for errors where commands could be executed with invalid or missing data.

Without basic assertions in place, commands can be executed with empty or invalid fields, leading to
inconsistent behavior or runtime exceptions. For example, a user may accidentally create a task with
an empty description or a missing deadline, which violates the expected behavior of the application.

Add assertions in the constructors of the Deadline, Event, Find, and ToDo commands to check that the
task description and other critical fields (like deadline for DeadlineCommand) are not empty. Additionally,
error handling logic is strengthened in the `executeCommand` method to throw a `CookieException` if validation fails.

Using assertions helps catch errors early during execution, ensuring that the commands always receive valid input.
Assertions are efficient because they validate conditions only during development/testing, while the error
handling in `executeCommand` ensures graceful handling in production.

This update improves code reliability and reduces the risk of handling incomplete or erroneous inputs.
While assertions are only active during testing, explicit error handling covers production use cases,
making the solution robust across environments.

Gilbb committed:
Merge pull request #4 from Gilbb/branch-A-Assertions

Add assertions to Deadline, Event, Find, and ToDo

Gilbb committed:
Add .yml file for CI

Gilbb committed:
Add javadoc comments for Classes

Gilbb committed:
Fix checkstyle errors

Gilbb committed:
Fix ordering of import statements

Gilbb committed:
Fix unit test assertion

Gilbb committed:
Fix unit test assertion for TaskList

Gilbb committed:
Merge pull request #5 from Gilbb/branch-A-CI

Branch a ci

Gilbb committed:
Add support for user-defined aliases

- Implemented functionality to allow users to define their own command aliases.
- Updated parser to resolve commands using aliases, improving user experience.
- Added error handling for invalid alias definitions and command resolution.

This update enhances the user interface by making command inputs more flexible
and customizable.

Gilbb committed:
Add support for users to view aliases

Gilbb committed:
Merge pull request #6 from Gilbb/branch-BCD-Extension

Branch bcd extension

Gilbb committed:
Fix build.gradle mainClass.set

Gilbb committed:
Add more detailed execption messages

Gilbb committed:
Merge pull request #7 from Gilbb/branch-DetailedExceptions

Add more detailed execption messages

Gilbb committed:
Add features list to README.md

Gilbb committed:
Add details for exiting to READme

Gilbb committed:
Fix minor typo

Gilbb committed:
Add error prevention for corrupted file

Gilbb committed:
Merge pull request #9 from Gilbb/branch-CorruptedFile

Add error prevention for corrupted file

Gilbb committed:
Save tasklist after every command

Gilbb committed:
Add exception handling for nfe
